ocean_lib/models/data_token.py

```diff
@@ -189,18 +189,17 @@ def get_start_order_logs(
         if consumer_address:
             topic1 = f"0x000000000000000000000000{consumer_address[2:].lower()}"
             topics = [topic0, None, topic1]
+        address = self.address if not from_all_tokens else None
 
-        filter_params = {"fromBlock": from_block, "toBlock": to_block, "topics": topics}
-        if not from_all_tokens:
-            # get logs only for this token address
-            filter_params["address"] = self.address
-
-        event_abi = self.events.OrderStarted().abi
-        logs = self.web3.eth.get_logs(filter_params)
-        parsed_logs = []
-        for lg in logs:
-            parsed_logs.append(get_event_data(self.web3.codec, event_abi, lg))
-        return parsed_logs
+        event_filter = EventFilter(
```

Comment:
The original logic is more similar to `getLogs` than to the `Filter` API.  Plus the `EventFilter` is never reused for polling. Please use `event.getLogs()` instead of `EventFilter`

Reply:
I agree with you @DMats here, but unfortunately I receive a weird error regarding `argument_filters`. Since there are `topics` in this function, I assume that I need to pass the list `topics` as a parameter for `getLogs`. It's strange that it works with `EventFilter` instead. A cause of this issue here may be the fact that `EventFilter` class takes the topics separately from `argument_filters`. As I saw in the [docs](https://web3py.readthedocs.io/en/stable/filters.html), the `argument_filters` param is a dict that can have `topics` included and it actually fails with `TypeError: getLogs() got multiple values for argument 'argument_filters'`.

Reply:
The `ContractEvent.getLogs()` method doesn't support a separate `topics` parameter.  Thus, `topics` need to be passed as a key/value pair in the `argument_filters` dict argument.

Reply:
I know. It was another problem.